engine/src/main/java/com/datatorrent/stram/util/ConfigUtils.java

```diff
@@ -68,15 +68,30 @@ public static String getRMUsername(Configuration conf)
     return principal;
   }
 
-  public static String getSchemePrefix(YarnConfiguration conf)
+  public static boolean isSSLEnabled(Configuration conf)
+  {
+    if (HttpConfig.Policy.HTTPS_ONLY == HttpConfig.Policy.fromString(
+        conf.get(YarnConfiguration.YARN_HTTP_POLICY_KEY, YarnConfiguration.YARN_HTTP_POLICY_DEFAULT))) {
+      return true;
+    }
+    return false;
+  }
+
+  public static String getSchemePrefix(Configuration conf)
   {
-    if (HttpConfig.Policy.HTTPS_ONLY == HttpConfig.Policy.fromString(conf.get(YarnConfiguration.YARN_HTTP_POLICY_KEY, YarnConfiguration.YARN_HTTP_POLICY_DEFAULT))) {
+    if (isSSLEnabled(conf)) {
```

Comment:
@PramodSSImmaneni : In-fact I won't make any change in this file. I would just add following code in StreamingAppMasterService

`if ("https//:".equals.(ConfigUtils. getSchemePrefix(config))) {
     appMasterTrackingUrl = "https://" + appMasterTrackingUrl;
    }
`